packages/modules/src/modules/headers/SignInPromptHeader.tsx

```diff
@@ -0,0 +1,192 @@
+
+        if (benefitIndex === -1) {
+            setBenefitIndex(0);
+            return;
+        }
+
+        for (let i = 0; i < DOTS_COUNT; i++) {
+            const delay = i === 0 ? 0 : FADE_TIME_MS + ANIMATION_DELAY_MS;
+            // Fade in individual dots
+            queueAnimation(() => {
+                setDotsVisible(currentState => {
+                    const newState = [...currentState];
+                    newState.splice(i, 1, true);
+                    return newState;
+                });
+            }, delay);
+        }
+
+        // Fade out all dots
+        queueAnimation(() => {
+            const newState = new Array(DOTS_COUNT).fill(false);
+            setDotsVisible(newState);
+        }, FADE_TIME_MS + ANIMATION_DELAY_MS);
+
+        // Fade in benefit text
+        queueAnimation(() => {
+            setBenefitVisible(true);
+        }, FADE_TIME_MS + ANIMATION_DELAY_MS);
+
+        if (benefitIndex < benefits.length - 1) {
+            // Fade out benefit text
+            queueAnimation(() => {
+                setBenefitVisible(false);
+            }, FADE_TIME_MS + TEXT_DELAY_MS);
+
+            // Trigger this effect to run again
+            queueAnimation(() => {
+                setBenefitIndex(benefitIndex + 1);
+            }, FADE_TIME_MS + ANIMATION_DELAY_MS);
+        }
+
+        const tick = () => {
+            const animationStep = animationSteps.shift();
+
+            if (!animationStep) {
+                return;
+            }
+
+            timeout = setTimeout(() => {
+                animationStep.callback();
+                tick();
+            }, animationStep.ms);
+        };
+
+        // Start this stage of the animation
+        tick();
+
+        return () => {
+            // Clear any timeouts still running in case of unexpected unmount
+            if (timeout) {
+                clearTimeout(timeout);
+            }
+        };
+    }, [benefitIndex]);
+
+    return (
+        <Hide below="tablet">
+            <h2 css={headingStyles}>{heading}</h2>
+            <h3 css={subHeadingStyles}>{subheading}</h3>
+
+            <div css={benefitsWrapper}>
+                <div css={dotsWrapper}>
+                    {dotsVisible.map((dotVisible, index) => {
+                        const dotCss = [dotStyles, fadeable];
+
+                        if (dotVisible) {
+                            dotCss.push(visible);
+                        }
+
+                        return <div css={dotCss} key={index} />;
+                    })}
+                </div>
+                <div css={benefitCss}>
+                    <div css={dotStyles} />
+                    <span css={benefitTextStyles}>{benefitText}</span>
+                </div>
+            </div>
+
+            {primaryCta && (
+                <ThemeProvider theme={buttonBrand}>
+                    <LinkButton priority="primary" href={primaryCta.ctaUrl} size="xsmall">
+                        {primaryCta.ctaText}
```

Comment:
does it need to handle `mobileContent`?

Reply:
We don't have a use case for that yet in the designs.